File: OrderSale/src/main/java/com/micro/OrderSale/model/Stock.java
package com.micro.OrderSale.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Stock {
    private Long productId;
    private String category_name;
    private String product_name;
    private int product_price;
    private Long purchase;
    private Long sale;
    private Long available;
}

File: InventoryClient/src/main/webapp/js/stock.js
//fetching stock
const API_FOR_SUPPLIER_STOCK_PORT = "http://localhost:8888/supplierstock";
function stocktable() {
  fetch(API_FOR_SUPPLIER_STOCK_PORT + "/stock", {
    method: "GET",
  })
    .then((response) => response.json())
    .then((data) => {
      const stocktable = document.getElementById("stockTable");
      // console.log(data);
      stocktable.innerHTML = "";
      if (Array.isArray(data)) {
        data.forEach((stock) => {
          stocktable.innerHTML +=
            "<tr>" +
            "<td>" +
            stock.category_name +
            "</td>" +
            "<td>" +
            stock.product_name +
            "</td>" +
            "<td>" +
            stock.product_price +
            "</td>" +
            "<td>" +
            stock.purchase +
            "</td>" +
            "<td>" +
            stock.sale +
            "</td>" +
            "<td>" +
            stock.available +
            "</td>" +
            "</tr>";
        });
      }
    })
    .catch((error) => console.error(error));
}

stocktable();
